### PR TITLE
[minor] Add JDBC configuration to connect to external databases

### PR DESCRIPTION
**Pull request by sbudhira**
1. Added JDBC configuration to connect to external databases ( DB2, Oracle, SQL Server) for SSL/non-SSL
2. Added environment variables for tablespace and indexspace names : MAS_APP_SETTINGS_INDEXSPACE and  MAS_APP_SETTINGS_TABLESPACE

Tested with SSL and Non-SSL database
Tested with DB2, Oracle and SQL Server databases 

[jdbccfg.zip](https://github.com/ibm-mas/ansible-devops/files/10394090/jdbccfg.zip)

The zip file contains JDBC configuration files for Db2, Oracle, SQL Server (SSL) generated during oneclick_add_manage playbook execution.

### CODE REVIEW

**Review by alequint — CHANGES_REQUESTED** [review]

File: ibm/mas_devops/playbooks/oneclick_add_manage.yml [hunk 1]
```diff
@@ -19,6 +19,14 @@
...
+    configure_external_db: "{{ lookup('env', 'CONFIGURE_EXTERNAL_DATABASE') | default('False', true) | bool }}"
+
+    mas_config_scope: "{{ lookup('env', 'MAS_CONFIG_SCOPE') | default('system', true) }}" # Supported values are "system", "ws", "app", or "wsapp"
+
+    # Check if SSL is enabled
+    ssl_enabled: "{{ lookup('env', 'SSL_ENABLED') | default('True', true) | bool }}"
+
   pre_tasks:
     # For the full set of supported environment variables refer to the playbook documentation
     - name: Check for required environment variables
```
[the remark this answers is not in the document]
Comment (sbudhira) [reply ibm/mas_devops/playbooks/oneclick_add_manage.yml:32]: @alequint These assertion are done in the gencfg_jdbc.

**Review by alequint — CHANGES_REQUESTED** [review]

File: ibm/mas_devops/playbooks/oneclick_add_manage.yml [hunk 2]
```diff
@@ -29,9 +37,36 @@
...
+      assert:
+        that:
+          - lookup('env', 'DB_INSTANCE_ID') != ""
+          - lookup('env', 'MAS_JDBC_USER') != ""
+          - lookup('env', 'MAS_JDBC_PASSWORD') != ""
+          - lookup('env', 'MAS_JDBC_URL') != ""
+          - lookup('env', 'MAS_APP_SETTINGS_DB2_SCHEMA') != ""
+          - lookup('env', 'MAS_APP_SETTINGS_TABLESPACE') != ""
+          - lookup('env', 'MAS_APP_SETTINGS_INDEXSPACE') != ""
+        fail_msg: "One or more JDBC configuration environment variables are not defined"
```
[the remark this answers is not in the document]
Comment (sbudhira) [reply ibm/mas_devops/playbooks/oneclick_add_manage.yml]: updated message

**Review by alequint — APPROVED** [review]

**Comment by alequint** [comment]
Looks good to me now @sbudhira and @durera

File: docs/changes.md [hunk 3]
```diff
@@ -1,6 +1,8 @@
 ## Changes
 Note that links to pull requests prior to public release of the code (4.0) direct to IBM GitHub Enterprise, and will only be accessible to IBM employees.
 
+
+- `12.9` Added JDBC configuration to connect to an external SSL and not SSL enabled database for Manage application (DB2, Oracle and SQL Server) ([#600]( https://github.com/ibm-mas/ansible-devops/pull/600))
```
Comment (durera) [inline docs/changes.md:5]: Entry in changelog should match the PR title `Add JDBC configuration to connect to external databases`, also needs adding to the duplicate changelog that shows up in ansible galaxy (see `ibm/mas_devops/README.md`)

**Review by durera — CHANGES_REQUESTED** [review]

File: docs/changes.md (hunk 3 above)
Comment (sbudhira) [reply docs/changes.md]: Updated changelogs.

**Review by alequint — APPROVED** [review]

**Review by durera — APPROVED** [review]

**Pull request merged**